core/balances.go

```diff
@@ -184,7 +184,8 @@ func (bc *BaseContract) TokenBalanceAddWithTicker(
 	reason string,
 ) error {
 	if stub, ok := bc.GetStub().(*cachestub.TxCacheStub); ok {
-		stub.AddAccountingRecord(bc.config.GetSymbol(), address, &types.Address{}, amount, reason)
+		accountingToken := getAccountingToken(ticker, bc.config.GetSymbol())
+		stub.AddAccountingRecord(accountingToken, address, &types.Address{}, amount, reason)
```

Comment:
If a symbol is a prefix of a ticker, I think we should check whether it is a prefix of a ticker. If it is not, we should return an error.

I also ask you to add unit tests.